### PR TITLE
Add graphql directives bbe

### PR DESCRIPTION
**Pull request by MohamedSabthar**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: examples/graphql-directives/graphql_directives.bal [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+// Marking enum value as deprecated
+public enum Gender {
+    MALE="Male",
+    FEMALE="Female",
+    # # Deprecated
+    # The `NON_BINARY` is deprecated use `OTHER` instead
+    @deprecated
+    NON_BINARY="Non-binary",
+    OTHER="Other"
```
[the remark this answers is not in the document]
Comment (MohamedSabthar) [reply examples/graphql-directives/graphql_directives.bal]: Updated

File: examples/graphql-directives/graphql_directives.client.out [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+curl -X POST -H "Content-type: application/json" -d '{ "query": "{ profile { name, address @skip(if: true) { city } } }" }' 'http://localhost:4000/graphql'
+{"data":{"profile":{"name":"Walter White"}}}
+
+
+# Using @include directive in query
+# The query used: { profile { name, address @include(if: true) { city } } }
+
+curl -X POST -H "Content-type: application/json" -d '{ "query": "{ profile { name, address @include(if: true) { city } } }" }' 'http://localhost:4000/graphql'
+{"data":{"profile":{"name":"Walter White", "address":{"city":"Albuquerque"}}}}
```
Comment (ThisaruGuruge) [inline examples/graphql-directives/graphql_directives.client.out:12]: Shall we add an introspection query with deprecated fields and enum values?

File: examples/graphql-directives/graphql_directives.bal [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    Address address;
+};
+
+public type Address record {
+    string number;
+    string street;
+    string city;
+};
+
+// Marking enum value as deprecated
```
Comment (praneesha) [inline examples/graphql-directives/graphql_directives.bal:16]:
```suggestion
// Marking enum value as deprecated.
```

File: examples/graphql-directives/graphql_directives.bal [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+public enum Gender {
+    MALE,
+    FEMALE,
+    # # Deprecated
+    # The `NON_BINARY` is deprecated use `OTHER` instead
```
Comment (praneesha) [inline examples/graphql-directives/graphql_directives.bal:21]:
```suggestion
    # The `NON_BINARY` is deprecated. Use `OTHER` instead.
```

File: examples/graphql-directives/graphql_directives.bal [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    gender: MALE,
+    address: {
+        number: "308",
+        street: "Negra Arroyo Lane",
+        city: "Albuquerque"
+    }
+};
+
+service /graphql on new graphql:Listener(4000) {
+    // Marking a field as deprecated
```
Comment (praneesha) [inline examples/graphql-directives/graphql_directives.bal:39]:
```suggestion
    // Marking a field as deprecated.
```

File: examples/graphql-directives/graphql_directives.bal [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    # # Deprecated
+    # The `person` field is deprecated use `profile` instead
```
Comment (praneesha) [inline examples/graphql-directives/graphql_directives.bal:41]:
```suggestion
    # The `person` field is deprecated. Use `profile` instead.
```

File: examples/graphql-directives/graphql_directives.client.out [hunk 7]
```diff
@@ -0,0 +1,17 @@
+# Using @skip directive in query
```
Comment (praneesha) [inline examples/graphql-directives/graphql_directives.client.out:1]:
```suggestion
# Using the `@skip` directive in a query.
```

File: examples/graphql-directives/graphql_directives.client.out [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+# The query used: { profile { name, address @skip(if: true) { city } } }
+
+curl -X POST -H "Content-type: application/json" -d '{ "query": "{ profile { name, address @skip(if: true) { city } } }" }' 'http://localhost:4000/graphql'
+{"data":{"profile":{"name":"Walter White"}}}
+
+
+# Using @include directive in query
```
Comment (praneesha) [inline examples/graphql-directives/graphql_directives.client.out:8]:
```suggestion
# Using the `@include` directive in a query.
```

File: examples/graphql-directives/graphql_directives.client.out [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+# The query used: { profile { name, address @include(if: true) { city } } }
+
+curl -X POST -H "Content-type: application/json" -d '{ "query": "{ profile { name, address @include(if: true) { city } } }" }' 'http://localhost:4000/graphql'
+{"data":{"profile":{"name":"Walter White", "address":{"city":"Albuquerque"}}}}
+
+# Introspections to check deprecated enum values
```
Comment (praneesha) [inline examples/graphql-directives/graphql_directives.client.out:14]:
```suggestion
# Introspections to check deprecated enum values.
```

File: examples/graphql-directives/graphql_directives.md [hunk 10]
```diff
@@ -0,0 +1,20 @@
+# Directives
+
+The Ballerina GraphQL supports the following graphql directives.
```
Comment (praneesha) [inline examples/graphql-directives/graphql_directives.md:3]:
```suggestion
The Ballerina GraphQL supports the following GraphQL directives.
```

File: examples/graphql-directives/graphql_directives.md [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+
+`@depricated` - Marks the schema definition of a field or enum value as deprecated with an optional reason.
+`@skip(if: Boolean!)` -  	If true, the decorated field or fragment in an operation is not resolved by the GraphQL server.
+`@include(if: Boolean!)` - If false, the decorated field or fragment in an operation is not resolved by the GraphQL server.
+
+The following example demonstrates the use of graphql directives
```
Comment (praneesha) [inline examples/graphql-directives/graphql_directives.md:9]:
```suggestion
The following example demonstrates the use of GraphQL directives.
```

File: examples/graphql-directives/graphql_directives.md [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+
+::: code directives.bal :::
+
+Run the service by executing the following command.
+
+::: out graphql_directives.server.out :::
+
+Following requests demonstrate the usage of `@skip` and `@include` directives
```
Comment (praneesha) [inline examples/graphql-directives/graphql_directives.md:17]:
```suggestion
The requests below demonstrate the usage of the `@skip` and `@include` directives.
```

File: examples/graphql-directives/graphql_directives.md [hunk 13]
```diff
@@ -0,0 +1,20 @@
...
+Invoke the service using the following curl command to inspect the result.
```
Comment (praneesha) [inline examples/graphql-directives/graphql_directives.md:18]:
```suggestion
Invoke the service using the following cURL command to inspect the result.
```

**Review by ThisaruGuruge — APPROVED** [review]

**Review by praneesha — APPROVED** [review]

**Pull request merged**